### PR TITLE
Travis: port changes from SVT-AV1

### PR DESCRIPTION
**Pull request by 1480c1**
Maybe might be better to add a debug build like svt-av1 so that if there are leaks, it would be easier to spot.

Make sure to uncomment out the last line for MacOS once it's confirmed buildable.

### CODE REVIEW

**Comment by tianjunwork** [comment]
Hi @1480c1, there are more update in AV1 travis. E.g. basic encoding, and ffmpeg/gst plugin. Would you like to add the support to HEVC too? Thanks!

**Comment by 1480c1** [comment]
Sure, I can update to have similar parity to svt-av1. Problem will still be with macOS

**Comment by tianjunwork** [comment]
@1480c1 macOS support is low priority to SVT, since its main usage is on server.

**Comment by 1480c1** [comment]
@tianjunwork or @jsunintel, does SvtHevcEncApp support stdin from ffmpeg properly?
When I tried `ffmpeg -i akiyo_cif.y4m -nostdin -f rawvideo -pix_fmt yuv420p - | ./SvtHevcEncApp -i stdin -w 352 -h 288 -fps-num 30000 -fps-denom 1001 -b test2.h265`
It kept on encoding and didn't even stop when it reached 2000 frames. Looking at the test2.h265, it stopped on the last frame. Does encoding through stdin require limiting frames?

```
4.18.0-22-generic #23~18.04.1-Ubuntu SMP x86_64 GNU/Linux
Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
ffmpeg version 3.4.6-0ubuntu0.18.04.1
SVT-HEVC build on master branch
```

**Comment by 1480c1** [comment]
Maybe it's fine? When doing `./SvtHevcEncApp -i stdin -w 352 -h 288 -fps-num 30000 -fps-denom 1001 -b test2.h265 < akiyo_cif.y4m`, it properly stops at 300 frames

**Comment by 1480c1** [comment]
Also, should I add the python test found in https://github.com/OpenVisualCloud/SVT-HEVC/tree/master/Tests and a decoder test?

**Comment by tianjunwork** [comment]
Yes.The application also supports reading from pipe. E.g. ffmpeg -i [input.mp4] -nostdin -f rawvideo -pix_fmt yuv420p - | SvtHevcEncApp.exe -i stdin -n [number_of_frames_to_encode] -w [width] -h [height]

**Comment by tianjunwork** [comment]
https://github.com/OpenVisualCloud/SVT-HEVC/tree/master/Tests includes all the tests for our backend Jenkins CI. It runs for every PR. Since the tests are already covered, Travis can be quick&basic test on build and one encode/decode command.

**Comment by tianjunwork** [comment]
> The application also supports reading from pipe. E.g. ffmpeg -i [input.mp4] -nostdin -f rawvideo -pix_fmt yuv420p - | SvtHevcEncApp.exe -i stdin -n [number_of_frames_to_encode] -w [width] -h [height]
 -n [number_of_frames_to_encode] is a must when you use "-i stdin"

**Comment by 1480c1** [comment]
ccache will need at least one run before it actually works.

Probably ready for review now

File: .travis.yml [hunk 1]
```diff
@@ -1,54 +1,186 @@
...
-     
+  # valgrind --tool=memcheck --leak-check=yes --show-reachable=yes ./SvtHevcEncApp -help
+  # Coding style check
+    - name: Style check
+      stage: style
+      addons: skip
+      before_install: skip
+      install: skip
+      script:
+        - git grep -I -n -P "\t|\r| $" -- .  && IS_FAIL=true || true
```
Comment (tianjunwork) [inline .travis.yml]: What is this line checking? Only "\t\r"? The travis log exceeds the max length.

Comment (1480c1) [reply .travis.yml]: tabs and carriage return and trailing spaces

Comment (1480c1) [reply .travis.yml]: I believe that https://github.com/OpenVisualCloud/SVT-HEVC/pull/226 should fix most of the problems, although maybe not all

Comment (1480c1) [reply .travis.yml]: I have made it a little less verbose, although, the problem files still remain

File: Config/SVTQPFile.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright(c) 2018 Intel Corporation 
+# Copyright(c) 2018 Intel Corporation
```
Comment (tianjunwork) [inline Config/SVTQPFile.txt]: This is changed by accident?

Comment (1480c1) [reply Config/SVTQPFile.txt]: Yes, going to undo, just a moment

**Comment by tianjunwork** [comment]
@1480c1 Ready to go?

**Comment by 1480c1** [comment]
Yea

**Comment by 1480c1** [comment]
Actually, got to add my changes from the pr in SVT-AV1

**Comment by 1480c1** [comment]
Done

**Comment by tianjunwork** [comment]
Thanks for your contribution again👍

**Pull request merged**